sdk/go/common/apitype/search.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apitype
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type ResourceSearchResponse struct {
+	Total        *int64                    `json:"total,omitempty"`
+	Resources    []ResourceResult          `json:"resources,omitempty"`
+	Aggregations map[string]Aggregation    `json:"aggregations,omitempty"`
+	Pagination   *ResourceSearchPagination `json:"pagination,omitempty"`
+}
+
+// ResourceResult is the user-facing type for our indexed resources.
+//
+// If you add a property here, don't forget to update fieldMappings to make it
+// queryable!
+type ResourceResult struct {
+	Created      *string                      `json:"created,omitempty" csv:"created"`
+	Custom       *bool                        `json:"custom,omitempty" csv:"custom"`
+	Delete       *bool                        `json:"delete,omitempty" csv:"delete"`
+	Dependencies []string                     `json:"dependencies,omitempty" csv:"-"`
+	ID           *string                      `json:"id,omitempty" csv:"id"`
+	Matches      map[string][]json.RawMessage `json:"matches,omitempty" csv:"-"`
+	Modified     *string                      `json:"modified,omitempty" csv:"modified"`
+	Module       *string                      `json:"module" csv:"module"`
+	Name         *string                      `json:"name,omitempty" csv:"name"`
+	Package      *string                      `json:"package" csv:"package"`
+	Parent       struct {
+		URN string `json:"urn"`
+	} `json:"parent"`
```

Comment:
```suggestion
	ParentURN *string `json:"parent.urn"
```

Reply:
I was told nested struct tags like this won't work - is that not the case?

Reply:
No nesting involved, the key is just "parent.urn" verbatim, e.g.
```
      "parent.urn": "urn:pulumi:...",
      ...
      "provider.urn": "urn:pulumi:...",
```

Reply:
cc: @abhinav - I was asked to change this because I was told this syntax is not valid.

Reply:
Aha! That makes sense. A bit strange, but I'll update appropriately :)